### ISSUE TITLE
Using innerHTML could break things

### ISSUE THREAD

**Issue by Tymewalk** [Tymewalk/Tymoji#23]
This already broke the "Add to Backpack" block. We need to find a new way to edit Tymojis in so things don't break.

**Comment by joker314** [comment]
Well, how about we change the system so you type the tymojis as
`[img]slight-smile[/img]`
and we just chenge the src of those invalid images. This has the bonus that those without the extension will understand that it is an image and not random text within tildas.

**Comment by CosmicWebServices** [comment]
That seems complex

**Comment by joker314** [comment]
The user just has a shorthand for linking to certain images..

I mean, I guess it is.

The trouble with the backpack button was that event listeners were being set to the backpack buttons. Then Tymoji would replace the innerHTML and thus create new elements. Since the event listeners were listening to the old buttons, the requests weren't being logged.

==An idea below======
We have to wrap these things into tags. The way we can do this is as follows: when you have the extension and you press the submit button the post does not yet submit. Instead, tymoji adds `[color]~slight-smile~[/color]` into the field. This makes no difference to the output both with and without the extension, but when Tymoji goes to replace the slight smile it first sinks into the `style` element controlling the colour. This way only the text that needs to be replaced gets replaced. AND it breaks all existing Tymoticons. UNLESS we have Tymoji also check for posts with the old method, and ONLY those posts get broken by features the ST may decide to add. Any non-Tymoticon posts will not get broken, and all new Tymoticon posts will also be safe.

Hey, while we are using this idea, why not instead of using colour, use links to the tymoticon thread. If you make a Tymojicon in the new version, it could be wrapped before it gets submitted like so:
`[color=red][url=LINK TO TYMOJI THREAD]~slight-smile~[/url][/color]`
and now people without Tymoji can get Tymoji.

### tl;dr new posts get optimised for `innerHTML` not breaking things, old posts *with* Tymoji should use the current `innerHTML` method (but it should be isolated to only those posts), and people without Tymoji will see tymojicons as red links to the thread (and the text of the tymojicon)

**Comment by joker314** [comment]
Does anybody think this is a good way of handling the issue>
